examples/blue-green-upgrade/modules/eks_cluster/main.tf

```diff
@@ -172,200 +256,388 @@ data "aws_caller_identity" "current" {}
+  cluster_version                = local.cluster_version
+  cluster_endpoint_public_access = true
 
-  # EKS CONTROL PLANE VARIABLES
-  cluster_version = local.cluster_version
+  vpc_id     = data.aws_vpc.vpc.id
+  subnet_ids = data.aws_subnets.private.ids
+
+  #we uses only 1 security group to allow connection with Fargate, MNG, and Karpenter nodes
+  create_node_security_group = false
+  eks_managed_node_groups = {
+    initial = {
+      node_group_name = local.node_group_name
+      instance_types  = ["m5.large"]
+
+      min_size     = 1
+      max_size     = 5
+      desired_size = 3
+      subnet_ids   = data.aws_subnets.private.ids
+    }
+  }
 
-  # List of map_roles
-  map_roles = [
+  manage_aws_auth_configmap = true
+  aws_auth_roles = flatten([
+    module.eks_blueprints_admin_team.aws_auth_configmap_role,
+    [for team in module.eks_blueprints_dev_teams : team.aws_auth_configmap_role],
     {
-      rolearn  = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${var.eks_admin_role_name}" # The ARN of the IAM role
-      username = "ops-role"                                                                                    # The user name within Kubernetes to map to the IAM role
-      groups   = ["system:masters"]                                                                            # A list of groups within Kubernetes to which the role is mapped; Checkout K8s Role and Rolebindings
+      rolearn  = module.karpenter.role_arn
+      username = "system:node:{{EC2PrivateDNSName}}"
+      groups = [
+        "system:bootstrappers",
+        "system:nodes",
+      ]
+    },
+    {
+      rolearn  = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${local.eks_admin_role_name}" # The ARN of the IAM role
+      username = "ops-role"                                                                                      # The user name within Kubernetes to map to the IAM role
+      groups   = ["system:masters"]                                                                              # A list of groups within Kubernetes to which the role is mapped; Checkout K8s Role and Rolebindings
     }
+  ])
+
+  tags = merge(local.tags, {
+    # NOTE - if creating multiple security groups with this module, only tag the
+    # security group that Karpenter should utilize with the following tag
+    # (i.e. - at most, only one security group should have this tag in your account)
+    "karpenter.sh/discovery" = "${local.environment}-${local.service}"
+  })
+}
+
+module "eks_blueprints_admin_team" {
+  source  = "aws-ia/eks-blueprints-teams/aws"
+  version = "~> 0.2"
+
+  name = "admin-team"
+
+  enable_admin = true
+  users = [
+    data.aws_caller_identity.current.arn,
+    "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:user/${local.iam_platform_user}",
+    "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${local.eks_admin_role_name}"
```

Comment:
Are we creating this user?

Add the `data_source` for partition.
```  "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:role/${local.eks_admin_role_name}"
```

Reply:
We don't create the platform_user, in the terraform.tfvars we can specify the name of the user to be added to the platform team.

What do you mean by add the data_source for partition ?